tekton/predicates.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tekton
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+)
+
+// PipelineRunSucceededPredicate returns a predicate which filters out all objects except
+// internal requests PipelineRuns which have just succeeded.
+func PipelineRunSucceededPredicate() predicate.Predicate {
```

Comment:
may be nice to have something about internal requests in the name but that is just a nit pick, no biggie

Reply:
Renamed to `InternalRequestPipelineRunSucceededPredicate`.